Kernel/Arch/x86_64/init.cpp

```diff
@@ -177,12 +181,26 @@ extern "C" [[noreturn]] UNMAP_AFTER_INIT void init(BootInfo const& boot_info)
     // We need to copy the command line before kmalloc is initialized,
     // as it may overwrite parts of multiboot!
     CommandLine::early_initialize(kernel_cmdline);
-    memcpy(multiboot_copy_boot_modules_array, multiboot_modules, multiboot_modules_count * sizeof(multiboot_module_entry_t));
-    multiboot_copy_boot_modules_count = multiboot_modules_count;
 
     new (&bsp_processor()) Processor();
     bsp_processor().early_initialize(0);
 
+    if (multiboot_modules_count > 0) {
+        dbgln("Initramfs image {}, end {}", multiboot_module_ramdisk_physical_start, multiboot_module_ramdisk_physical_end);
+        if (multiboot_module_ramdisk_physical_start > PhysicalAddress(0xffffffff) || multiboot_module_ramdisk_physical_end > PhysicalAddress(0xffffffff)) {
+            // FIXME: Add support for using initramfs images in such memory regions.
+            dbgln("Detected initramfs higher than 4GB, ignoring");
+        } else {
+            multiboot_copy_boot_modules_array[0].start = multiboot_module_ramdisk_physical_start.get();
+            multiboot_copy_boot_modules_array[0].end = multiboot_module_ramdisk_physical_end.get();
+            if (multiboot_module_ramdisk_physical_string_addr <= PhysicalAddress(0xffffffff)) {
+                multiboot_copy_boot_modules_array[0].string_addr = multiboot_module_ramdisk_physical_string_addr.get();
+            }
+        }
+        if (multiboot_modules_count > 1)
+            dbgln("Detected more than one multiboot modules, ignoring the rest.");
+    }
+
```

Comment:
Might be nicer to factor this out into a function, such that we don't have 'freestanding' logic in the init function.

Reply:
Since multiboot1 is not effective on arm64 bare-metal, how would you abstract it in a sane manner?

Reply:
My thinking was that it was cluttering the init function, at least for now we don't need this for aarch64. So it can stay like this I guess.

Reply:
Yeah, I rather keep it this way now. If you send a patch to enable initramfs on aarch64, feel free to abstract it in a different way.